src/execution_impl.rs

```diff
@@ -24,17 +25,37 @@ fn create_instance(limit: u64, module: &[u8], env: &Env) -> Result<Instance> {
     // See https://docs.rs/wasmer-compiler-singlepass/latest/wasmer_compiler_singlepass/
     let mut compiler_config = Singlepass::new();
 
-    // NaNs are non-deterministic in WASM: https://github.com/WebAssembly/design/blob/main/Nondeterminism.md
-    // Ensure determinism by canonicalizing their representation
+    // Turning-off sources of potential non-determinism,
+    // see https://github.com/WebAssembly/design/blob/037c6fe94151eb13e30d174f5f7ce851be0a573e/Nondeterminism.md
+
+    // Turning-off in the compiler:
+
+    // Canonicalize NaN.
     compiler_config.canonicalize_nans(true);
 
+    // Enable stack check.
+    compiler_config.enable_stack_check(true);
```

Comment:
Actually it's kind of interesting, I see nothing in the code actually using that flag, see https://github.com/wasmerio/wasmer/blob/8f2e49d52823cb7704d93683ce798aa84b6928c8/lib/compiler-singlepass/src/codegen_x64.rs (compare for example with the use of `config.enable_nan_canonicalization`)

Reply:
I think what was done here may point us in the right direction https://github.com/wasmerio/wasmer/pull/2807

I'll remove this flag because it seems useless and it may give us some false confidence that we're actually addressing the problem.